src/mapDataManipulation/manipulateGain.ts

```diff
@@ -0,0 +1,46 @@
+import { mapDataManipulation } from 'src/mapDataManipulation/mapDataManipulation';
+
+// The algorithm works with numbers between 0 and 1, so we must:
+// - change the interval of the values from [0, 255] to [0, 1]
+// - change the values according to the algorithm
+// - change the interval of the values from [0, 1] back to [0, 255]
+
+// separate algorithm for manipulating gain
+export async function manipulateGain(
+  originalBlob: Blob,
+  newGain: number,
+  newMinValue?: number,
+  newMaxValue?: number,
+  newOffset?: number,
+): Promise<Blob> {
+  const minValue = newMinValue ? newMinValue : 0.0;
+  const maxValue = newMaxValue ? newMaxValue : 1.0;
+  const gain = newGain ? newGain : 1.0;
+  const factor = gain / (maxValue - minValue);
+  let offset = newOffset ? newOffset : 0.0;
+  offset = offset - factor * minValue;
+
+  let newRedValues = [...Array(256).keys()];
+  let newGreenValues = [...Array(256).keys()];
+  let newBlueValues = [...Array(256).keys()];
+
+  // change the interval of the values from [0, 255] to [0, 1]
+  newRedValues = newRedValues.map(x => (1 / 255) * x);
+  newGreenValues = newGreenValues.map(x => (1 / 255) * x);
+  newBlueValues = newBlueValues.map(x => (1 / 255) * x);
+
+  // change the values according to the algorithm
+  newRedValues = newRedValues.map(x => Math.max(0.0, x * factor + offset));
+  newGreenValues = newGreenValues.map(x => Math.max(0.0, x * factor + offset));
+  newBlueValues = newBlueValues.map(x => Math.max(0.0, x * factor + offset));
+
+  // change the interval of the values from [0, 1] back to [0, 255]
+  newRedValues = newRedValues.map(x => Math.round((255 / 1) * x));
+  newGreenValues = newGreenValues.map(x => Math.round((255 / 1) * x));
+  newBlueValues = newBlueValues.map(x => Math.round((255 / 1) * x));
+
+  const manipulatePixel = function(r: number, g: number, b: number, a: number): object {
+    return { r: newRedValues[r], g: newGreenValues[g], b: newBlueValues[b], a };
+  };
+  return await mapDataManipulation(originalBlob, manipulatePixel);
```

Comment:
Since this code is used by `manupulateGain`, `manipulateGamma` and `manipulateGainGamma` I think we should write a function that can be reused in all 3 files.

Reply:
I refactored / rewrote the things quite a bit, but used that advice while at it.

Reply:
resolving so i can merge